pkg/types/types.go

```diff
@@ -261,10 +265,28 @@ func MergeMetrics(metrics [][]Metric, replicaMismatchConfig cfg.RenderReplicaMis
 			zap.Int("replica_points_total", metricsStat.DataPointCount),
 		)
 	}
+	if len(diffResolutionMetrics) > 0 {
+		logger.Info("metric with different schema observed",
+			zap.Strings("schema_mismatch_metrics", diffResolutionMetrics),
+			zap.Int("metrics total", len(diffResolutionMetrics)),
+		)
+	}
 
 	return merged, metricsStat
 }
 
+func isStepTimeMatching(ms []Metric) bool {
+	if len(ms) != 0 {
+		defaultStep := ms[0].StepTime
+		for _, mr := range ms {
```

Comment:
Since we will have the guarantee of `len(ms) > 1` with the above requested change, we can simplify this to something like this
```
for i := 1; i < len(ms); i++ {
    if mr.StepTime != mr[0].StepTime {
        return false
        ...
```

Reply:
@grzkv : I'm not an expert here, but is variable (`defaultStep`) comparison in loop not better then comparison with structure element of slice (`mr[0].StepTime`)? Or compiler optimise that anyway?

Reply:
@deniszh Interesting question! There's no difference, I think. Both will be a single address in assembly. The compiler _may_ need to re-calculate the address in the second case, but I'm not sure, it may also optimize it away. Even if it does not, the difference is zero in practice.

Reply:
changed, thanks